buildkite/pipelines/bacalhau-golang.yaml

```diff
@@ -1,64 +1,64 @@
-cache:
-  paths:
-    - "/golangci-lint-cache"
-    - "/vendor-cache"
-    - "/node-modules-cache"
-  size: 20g
-  name: "vendor-module-cache"
-
 steps:
   - label: ":golang: Lint"
     command: "./buildkite/scripts/lint.sh"
     key: "lint"
     agents:
       queue: "buildkite-hosted-linux-medium"
-      
+
   - group: ":package: Build Tarball"
     key: "build-tarball"
     steps:
       - label: ":golang: Build linux amd64"
         command: "./buildkite/scripts/build_tarball.sh linux amd64"
         agents:
           queue: "buildkite-hosted-linux-small"
-          
+
       - label: ":golang: Build linux arm64"
         command: "./buildkite/scripts/build_tarball.sh linux arm64"
         agents:
           queue: "buildkite-hosted-linux-small"
-          
+
       - label: ":golang: Build linux armv6"
         command: "./buildkite/scripts/build_tarball.sh linux armv6"
         agents:
           queue: "buildkite-hosted-linux-small"
-          
+
       - label: ":golang: Build linux armv7"
         command: "./buildkite/scripts/build_tarball.sh linux armv7"
         agents:
           queue: "buildkite-hosted-linux-small"
-          
+
       - label: ":golang: Build darwin amd64"
         command: "./buildkite/scripts/build_tarball.sh darwin amd64"
         agents:
           queue: "buildkite-hosted-linux-small"
-          
+
       - label: ":golang: Build darwin arm64"
         command: "./buildkite/scripts/build_tarball.sh darwin arm64"
         agents:
           queue: "buildkite-hosted-linux-small"
-          
+
       - label: ":golang: Build windows amd64"
         command: "./buildkite/scripts/build_tarball.sh windows amd64"
         agents:
           queue: "buildkite-hosted-linux-small"
-        
+
   - label: ":testengine: Unit Test"
     command: "./buildkite/scripts/test.sh unit"
     key: "unit-test"
     agents:
       queue: "buildkite-hosted-linux-large"
-      
+
   - label: ":testengine: Integration Test"
     command: "./buildkite/scripts/test.sh integration"
     key: "integration-test"
     agents:
       queue: "buildkite-hosted-linux-large"
+
+  - wait: ~
+
+  - label: ":rocket: Release CLI"
+    command: "./buildkite/scripts/release_cli.sh"
+    key: "release-cli"
+    if: build.tag != ""
```

Comment:
Help me understand this. We're only releasing if the build has a tag, but this pipeline is only triggered when committed right? Also what is the relationship between this step and the release pipeline? Some comments would be really helpful.

Reply:
So this pipeline gets triggered when one of the following condition 

`build.branch == "main" || !build.pull_request.draft || build.pull_request.labels includes "build/go" || build.pull_request.labels includes "build/golang" || build.tag != null` 

I ll add this as a comment :)

Reply:
I got rid of the release pipeline so that it does not confuse. This is a simpler way where we don't have to deal with artifact upload and artifact download in another job.